src/credentials/attestation_verifier/DeviceAttestationVerifier.h

```diff
@@ -226,20 +226,17 @@ class WellKnownKeysTrustStore
 class ArrayAttestationTrustStore : public AttestationTrustStore
 {
 public:
-    ArrayAttestationTrustStore(const ByteSpan * derCerts, size_t numCerts) : mDerCerts(derCerts), mNumCerts(numCerts) {}
+    ArrayAttestationTrustStore(const ByteSpan * const * derCerts, size_t numCerts) : mDerCerts(derCerts), mNumCerts(numCerts) {}
```

Comment:
So this now asks for an array of _pointers_ to ByteSpan, and the actual ByteSpans have to live somewhere outside the array.

Why is this change needed?  If we do need to make it, that might be better as a separate PR.

Reply:
The way we're currently creating the array for the single use case in the SDK is like this:
```
static const ByteSpan kTestPaaRoots[] = {
    TestCerts::sTestCert_PAA_FFF1_Cert,
    TestCerts::sTestCert_PAA_NoVID_Cert,
};
```
This is not constinit, because the ByteSpan objects live elsewhere and have to be copied.

Defining a named constinit ByteSpan object for each cert and then collecting them into a list seems like an ergonomic way to use this, so making the array contain a pointer to the ByteSpan seems better than e.g. creating the ByteSpan directly inside the roots array.

Reply:
> because the ByteSpan objects live elsewhere and have to be copied.

More precisely, they are extern and you can't have an extern constexpr, right?

But then what we are really doing is replacing non-const init with linker relocations to fix up the pointers, which is about the same in terms of startup time....

Reply:
It's easy to validate / enforce mechanically that there are no initializers though (via `-no_inits` and/or `constinit`), whereas seeing whether or not an initializer does a lot of work is difficult.

Reply:
OK.  So what I am wondering, @ksperling-apple is now that we are c++17 whether we can make the "public" bits in `src/credentials/tests/CHIPAttCert_test_vectors.h` be `inline constexpr` so that we don't have to worry about the extern thing at all....

Reply:
Hm, I guess that's an option, but having these large constants defined extern and have them in only one translation unit seems generally the right thing. We do seem to have a bunch of other `constexpr` constants that should actually be `inline constexpr`. I'll put up a separate PR for that.

Reply:
@ksperling-apple I think the issue is that ArrayAttestationTrustStore is public API.  So changing it if we don't have to is not great....

I agree that having the large constants be extern is generally the right thing; just trying to figure out a way to deal with this without changing public APIs.

Reply:
Fixed by moving `kTestAttestationTrustStoreRoots` (formerly `kTestPaaRoots`) into `CHIPAttCert_test_vectors.cpp` where the certificate spans themselves are defined.